storm-core/src/jvm/org/apache/storm/daemon/supervisor/RunAsUserContainer.java

```diff
@@ -0,0 +1,107 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RunAsUserContainer extends BasicContainer {
+    private static final Logger LOG = LoggerFactory.getLogger(RunAsUserContainer.class);
+
+    public RunAsUserContainer(int port, LocalAssignment assignment, Map<String, Object> conf, String supervisorId,
+            LocalState state, ResourceIsolationInterface resourceIsolationManager, boolean recover) throws IOException {
+        super(port, assignment, conf, supervisorId, state, resourceIsolationManager, recover);
+        if (Utils.isOnWindows()) {
+            throw new RuntimeException("ERROR: Windows doesn't support running workers as different users yet");
+        }
+    }
+
+    public RunAsUserContainer(String workerId, Map<String, Object> conf, String supervisorId,
+            ResourceIsolationInterface resourceIsolationManager) throws IOException {
+        super(workerId, conf, supervisorId, resourceIsolationManager);
+        if (Utils.isOnWindows()) {
+            throw new RuntimeException("ERROR: Windows doesn't support running workers as different users yet");
+        }
+    }
+
+    private void signal(long pid, int signal) throws IOException {
+        List<String> commands = Arrays.asList("signal", String.valueOf(pid), String.valueOf(signal));
+        String user = getWorkerUser();
+        String logPrefix = "kill -"+signal+" " + pid;
+        SupervisorUtils.processLauncherAndWait(_conf, user, commands, null, logPrefix);
+    }
+    
+    @Override
+    protected void kill(long pid) throws IOException {
+        signal(pid, 15);
+    }
+    
+    @Override
+    protected void forceKill(long pid) throws IOException {
+        signal(pid, 9);
+    }
+    
+    @Override
+    protected boolean runProfilingCommand(List<String> command, Map<String, String> env, String logPrefix, File targetDir) throws IOException, InterruptedException {
+        String user = this.getWorkerUser();
+        String td = targetDir.getAbsolutePath();
+        LOG.info("Running as user:{} command:{}", user, command);
+        String containerFile = Utils.containerFilePath(td);
+        if (Utils.checkFileExists(containerFile)) {
+            SupervisorUtils.rmrAsUser(_conf, containerFile, containerFile);
+        }
+        String scriptFile = Utils.scriptFilePath(td);
+        if (Utils.checkFileExists(scriptFile)) {
+            SupervisorUtils.rmrAsUser(_conf, scriptFile, scriptFile);
+        }
+        String script = Utils.writeScript(td, command, env);
+        List<String> args = Arrays.asList("profiler", td, script);
+        int ret = SupervisorUtils.processLauncherAndWait(_conf, user, args, env, logPrefix);
+        return ret == 0;
+    }
+
+    @Override
+    public List<String> updateCommandForIsolation(List<String> command) {
+        //With run as user we do the isolation before switching users, so don't modify the original
+        // command line
+        return command;
+    }   
```

Comment:
@revans2 , Let's move updateCommandForIsolation into launchWorkerProcess, as we discussed. It will clean up the code.